src/App/components/Settings.tsx

```diff
@@ -95,11 +95,11 @@ DirectoriesCard.propTypes = {
 export default function Settings(): JSX.Element {
   const { settings } = useStores();
   const { analytics } = useAnalytics();
-  const repoUrl = repository.url.replace(/\.git$/, '');
+  const repoUrl = repository.url.replace(/\.git$/, "");
   const resetOutputDir = (): void => settings.setOutputDirectory(DEFAULT_OUTPUT_DIRECTORY);
   React.useEffect(() => {
-    analytics.trackScreenview('Settings');
-  }, []);
+    analytics.trackScreenview("Settings");
+  }, [analytics]);
```

Comment:
Can you explain what this does?

Reply:
React's `useEffect()` hook takes an optional second parameter called "dependency array" to help it decide when to execute. It is good practice to include every dependency in this array, even if it is never expected to change. I did check to make sure `analytics` never changes -- if it somehow did, this would force a re-render (bad). 

In this case our console lint warning was: 
```
webpackHotDevClient.js:138 ./src/App/components/Settings.tsx
  Line 102:6:  React Hook React.useEffect has a missing dependency: 'analytics'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
  ```
  ```
webpackHotDevClient.js:138 ./src/App/index.tsx
  Line 31:6:  React Hook React.useEffect has a missing dependency: 'analyticsContext.analytics'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
  ```

see also https://stackoverflow.com/questions/56240067/accessing-context-from-useeffect

Does that help?

Reply:
Yep, thanks.